src/main/scala/tilelink/Parameters.scala

```diff
@@ -10,32 +10,59 @@ import freechips.rocketchip.util.{RationalDirection,AsyncQueueParams, groupByInt
 import scala.math.max
 import scala.reflect.ClassTag
 
-case class TLManagerParameters(
-  address:            Seq[AddressSet],
-  resources:          Seq[Resource] = Seq(),
-  regionType:         RegionType.T  = RegionType.GET_EFFECTS,
-  executable:         Boolean       = false, // processor can execute from this memory
-  nodePath:           Seq[BaseNode] = Seq(),
+case class TLSupportedSizes(
+  acquireT:   TransferSizes = TransferSizes.none,
+  acquireB:   TransferSizes = TransferSizes.none,
+  arithmetic: TransferSizes = TransferSizes.none,
+  logical:    TransferSizes = TransferSizes.none,
+  get:        TransferSizes = TransferSizes.none,
+  putFull:    TransferSizes = TransferSizes.none,
+  putPartial: TransferSizes = TransferSizes.none,
+  hint:       TransferSizes = TransferSizes.none,
+  probe:      TransferSizes = TransferSizes.none
+)
+
+class TLManagerParameters private(
+  // The data have a P (for private) suffix to differentiate them from the functions
+  addressP:            Seq[AddressSet],
+  resourcesP:          Seq[Resource] = Seq(),
+  regionTypeP:         RegionType.T  = RegionType.GET_EFFECTS,
+  executableP:         Boolean       = false, // processor can execute from this memory
+  nodePathP:           Seq[BaseNode] = Seq(),
   // Supports both Acquire+Release+Finish of these sizes
-  supportsAcquireT:   TransferSizes = TransferSizes.none,
-  supportsAcquireB:   TransferSizes = TransferSizes.none,
-  supportsArithmetic: TransferSizes = TransferSizes.none,
-  supportsLogical:    TransferSizes = TransferSizes.none,
-  supportsGet:        TransferSizes = TransferSizes.none,
-  supportsPutFull:    TransferSizes = TransferSizes.none,
-  supportsPutPartial: TransferSizes = TransferSizes.none,
-  supportsHint:       TransferSizes = TransferSizes.none,
-  userBits:           Seq[UserBits] = Nil,
+  supportedSizesP:      TLSupportedSizes = TLSupportedSizes(),
+  emissionSizesP:       TLSupportedSizes = TLSupportedSizes(),
```

Comment:
Why not call these: supports and emits?

Then the API looks like: client.supports.arithmetic instead of client.supportsSizesP.arithmetic

... and supports should to none (like it does), but emits should default to TransferSizes(1, 4096) ... the safe default